mivs/models.py

```diff
@@ -0,0 +1,205 @@
+    primary_contact = Column(Boolean, default=False)
+    first_name      = Column(UnicodeText)
+    last_name       = Column(UnicodeText)
+    email           = Column(UnicodeText)
+    cellphone       = Column(UnicodeText)
+
+
+class IndieGame(MagModel):
+    studio_id         = Column(UUID, ForeignKey('indie_studio.id'))
+    title             = Column(UnicodeText)
+    brief_description = Column(UnicodeText)
+    genres            = Column(MultiChoice(c.INDIE_GENRE_OPTS))
+    description       = Column(UnicodeText)       # 500 max
+    how_to_play       = Column(UnicodeText)       # 1000 max
+    link_to_video     = Column(UnicodeText)
+    link_to_game      = Column(UnicodeText)
+    password_to_game  = Column(UnicodeText)
+    code_type         = Column(Choice(c.CODE_TYPE_OPTS), default=c.NO_CODE)
+    code_instructions = Column(UnicodeText)
+    build_status      = Column(Choice(c.BUILD_STATUS_OPTS))
+    build_notes       = Column(UnicodeText)       # 500 max
+    shown_events      = Column(UnicodeText)
+    video_submitted   = Column(Boolean, default=False)
+    submitted         = Column(Boolean, default=False)
+    agreed_liability  = Column(Boolean, default=False)
+    agreed_showtimes  = Column(Boolean, default=False)
+    agreed_reminder1  = Column(Boolean, default=False)
+    agreed_reminder2  = Column(Boolean, default=False)
+    status            = Column(Choice(c.GAME_STATUS_OPTS), default=c.NEW, admin_only=True)
+    judge_notes       = Column(UnicodeText, admin_only=True)
+
+    codes = relationship('IndieGameCode', backref='game')
+    reviews = relationship('IndieGameReview', backref='game')
+    screenshots = relationship('IndieGameScreenshot', backref='game')
+
+    email_model_name = 'game'
+
+    @property
+    def email(self):
+        return self.studio.email
+
+    @property
+    def missing_steps(self):
+        steps = []
+        if not self.link_to_game:
+            steps.append('You have not yet included a link to where the judges can access your game')
+        if self.code_type != c.NO_CODE:
+            if not self.codes:
+                steps.append('You have not yet attached any codes to this game for our judges to use')
+            elif not any(code.unlimited_use for code in self.codes) and len(self.codes) < c.CODES_REQUIRED:
+                steps.append('You have not attached the {} codes you must provide for our judges'.format(c.CODES_REQUIRED))
+        if not self.agreed_showtimes:
+            steps.append('You must agree to showtimes, or something')  # TODO: figure out what this is
+        if not self.agreed_liability:
+            steps.append('You must check the box that agrees to our liability waiver (click the Edit link above)')
+        return steps
+
+    @property
+    def video_submittable(self):
+        return bool(self.link_to_video)
+
+    @property
+    def submittable(self):
+        return not self.missing_steps
+
+
```

Comment:
FWIW, I'm not a huge fan of storing files in the database.  I much prefer to put them on the filesystem and just have the database know where to find the file.  There are advantages and disadvantages to each approach of course.  I just figured I'd mention that in case anyone had any strong opinions on the subject or was curious.